### PR TITLE
Add hasflag enumextensions

### LINKED ISSUE #214: Add HasFlag to EnumExtensions

### PR DESCRIPTION
**Pull request by nazarenomanco**
Fix #214 
@mauroservienti please check if it is ok now, thanks! :smile:

### CODE REVIEW

File: src/net35/Test.Radical/Test.Radical.csproj [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
+    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
+    <ProjectGuid>{31CC206D-2014-4612-811E-82B198385343}</ProjectGuid>
+    <OutputType>Library</OutputType>
+    <AppDesignerFolder>Properties</AppDesignerFolder>
+    <RootNamespace>Test.Radical</RootNamespace>
+    <AssemblyName>Test.Radical</AssemblyName>
+    <TargetFrameworkVersion>v4.5</TargetFrameworkVersion>
```
Comment (mauroservienti) [inline src/net35/Test.Radical/Test.Radical.csproj]: Make sure that the test project targets fx 3.5

File: src/Radical.sln [hunk 2]
```diff
@@ -101,6 +103,22 @@ Global
...
+		{31CC206D-2014-4612-811E-82B198385343}.Debug|x86.ActiveCfg = Debug|Any CPU
+		{31CC206D-2014-4612-811E-82B198385343}.Debug|x86.Build.0 = Debug|Any CPU
+		{31CC206D-2014-4612-811E-82B198385343}.Release|Any CPU.ActiveCfg = Release|Any CPU
+		{31CC206D-2014-4612-811E-82B198385343}.Release|Any CPU.Build.0 = Release|Any CPU
+		{31CC206D-2014-4612-811E-82B198385343}.Release|ARM.ActiveCfg = Release|Any CPU
+		{31CC206D-2014-4612-811E-82B198385343}.Release|ARM.Build.0 = Release|Any CPU
+		{31CC206D-2014-4612-811E-82B198385343}.Release|x64.ActiveCfg = Release|Any CPU
+		{31CC206D-2014-4612-811E-82B198385343}.Release|x64.Build.0 = Release|Any CPU
+		{31CC206D-2014-4612-811E-82B198385343}.Release|x86.ActiveCfg = Release|Any CPU
+		{31CC206D-2014-4612-811E-82B198385343}.Release|x86.Build.0 = Release|Any CPU
```
Comment (mauroservienti) [inline src/Radical.sln]: Why so many build configuration? Can you please reduce them to the minimum required? E.g. remove the ARM stuff.

File: src/net35/Test.Radical/Extensions/EnumExtensionFx35Test.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            Value8 = 8
+        }
+
+        [TestMethod]
+        public void enumExtensions_HasFlag_checking_set_flag_return_true()
+        {
+            var testCase = FlaggedTestEnum.Value1 | FlaggedTestEnum.Value2;
+            var actual = testCase.HasFlag(FlaggedTestEnum.Value1);
+
+            actual.Should().Be.True();
```
Comment (mauroservienti) [inline src/net35/Test.Radical/Extensions/EnumExtensionFx35Test.cs]: Please use the standard .net asserts, we're trying to remove dependencies

File: src/net35/Test.Radical/Extensions/EnumExtensionFx35Test.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        }
+
+        [TestMethod]
+        public void enumExtensions_HasFlag_checking_not_set_flag_return_false()
+        {
+            var testCase = FlaggedTestEnum.Value1 | FlaggedTestEnum.Value2;
+            var actual = testCase.HasFlag(FlaggedTestEnum.Value8);
+
+            actual.Should().Be.False();
```
Comment (mauroservienti) [inline src/net35/Test.Radical/Extensions/EnumExtensionFx35Test.cs]: Please use the standard .net asserts, we're trying to remove dependencies

File: src/net35/Test.Radical/UnitTest1.cs [hunk 5]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
```
Comment (mauroservienti) [inline src/net35/Test.Radical/UnitTest1.cs]: Remove this class entirely

File: src/net35/Test.Radical/packages.config [hunk 6]
```diff
@@ -0,0 +1,4 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<packages>
+  <package id="SharpTestsEx" version="1.1.1" targetFramework="net45" />
```
Comment (mauroservienti) [inline src/net35/Test.Radical/packages.config]: Once the asserts are the standard ones remove this dependency as well. Thanks

**Comment by mauroservienti** [comment]
I did an initial review

**Comment by nazarenomanco** [comment]
Done everything... sorry :blush: 
I try to exclude as much build configuration as I can... let me know.

**Comment by mauroservienti** [comment]
Looks good.

**Comment by mauroservienti** [comment]
we'll skip version bumping since we already have pending features in the queue and as soon as we have #213 merged it will be automatic.

**Comment by nazarenomanco** [comment]
Perfect.After this I can improve the EnumToBoolean Converter

**Comment by mauroservienti** [comment]
@nazarenomanco please, raise an issue for the improvement in the correct repository, this is now available on MyGet as unstable, it will be released on Nuget soon. However if you want to start working on the `EnumToBoolean Converter` you can use the package from Myget, everything should be already configured in the repo to automatically reference it as well. If this is not the case ping me.

**Pull request merged**